template/upstream.c.erb

```diff
@@ -24,6 +25,10 @@ static void btx_dispatch_<%= e.name_sanitized %>(
              ->usr_data<%= e.args.map{ |s| s.name }.join_with_prefix(", ") %>);
   }
 
+  <% e.args_to_free.each do |s| %>
+  <%= "free(#{s.name});" %>
+  <% end %>
```

Comment:
It is a great Idea to free in the dispatcher, since it is here where we can know if the specifc event have arguments to free. We just need to consider that we can not access the freed arrays in matching callbacks that take place after regular callbacks since the arrays were already freed.

We have a nother option, use the same calbacks insfraestructure to be called per event at the end of the regular and matching callbacks, this guarantee we free at the end. I think this also can work in the Source, where we are not freeing yet.